update

```diff
@@ -104,233 +92,171 @@ function deleteIfExists (filePathRelativeToRoot) {
   })
 }
 
-function buildVersion (build) {
-  let version = build.version
-  if (build.prerelease && build.prerelease.length > 0) {
-    version += '-' + build.prerelease
-  }
-  if (build.build && build.build.length > 0) {
-    version += '+' + build.build
+async function batchedAsyncMap (values, batchSize, dir, oldList) {
+  const batchPromises = []
+
+  for (let i = 0; i < values.length; i += batchSize) {
+    batchPromises.push(pool.exec('workerMain', [dir, values.slice(i, i + batchSize), oldList]))
```

Comment:
Did you run any profiling to see how your changes improved the performance of the current code? The execution time of your implementation does not seem to have any significant improvement in my machine:
Current implementation: `702.21s user 16.80s system 106% cpu 11:15.54 total`
PR implementation: `670.77s user 17.92s system 104% cpu 11:01.69 total`

Maybe we could also move some of the I/O operations to the workers instead of only moving the hash operations? Like, we could have a worker per binary directory and then multiple child-workers per-directory (each one reading/processing a subset of files in each directory). The main worker thread in each directory would then aggregate the "child/file workers" results and write the respective `list.[json,js,txt]` files or something like that. What do you think?

I also run a quick profiling here, which seems to hint that your code is spending most of the time waiting for file descriptors to become ready. Not sure if this is actually the case though:

PR code:
```
[C++]:
   ticks  total  nonlib   name
   1987   71.3%   93.9%  epoll_pwait@@GLIBC_2.6
     21    0.8%    1.0%  __write@@GLIBC_2.2.5
     14    0.5%    0.7%  fwrite@@GLIBC_2.2.5
     12    0.4%    0.6%  recvmsg@@GLIBC_2.2.5
     11    0.4%    0.5%  std::basic_ostream<char, std::char_traits<char> >& std::__ostream_insert<char, std::char_traits<char> >(std::basic_ostream<char, std::char_traits<char> >&, char const*, long)
      9    0.3%    0.4%  std::ostream::sentry::sentry(std::ostream&)
      8    0.3%    0.4%  __libc_malloc@@GLIBC_2.2.5
      
      
[Summary]:
   ticks  total  nonlib   name
     37    1.3%    1.7%  JavaScript
   2078   74.5%   98.3%  C++
     72    2.6%    3.4%  GC
    673   24.1%          Shared libraries
  ```
  
Current code:
```
[C++]:
   ticks  total  nonlib   name
  18358    2.9%    4.9%  epoll_pwait@@GLIBC_2.6
    901    0.1%    0.2%  __dynamic_cast
    793    0.1%    0.2%  __cxxabiv1::__vmi_class_type_info::__do_dyncast(long, __cxxabiv1::__class_type_info::__sub_kind, __cxxabiv1::__class_type_info const*, void const*, __cxxabiv1::__class_type_info const*, void const*, __cxxabiv1::__class_type_info::__dyncast_result&) const
    711    0.1%    0.2%  __libc_malloc@@GLIBC_2.2.5

[Summary]:
   ticks  total  nonlib   name
   1923    0.3%    0.5%  JavaScript
  24553    3.9%    6.5%  C++
  12993    2.1%    3.5%  GC
  256700   40.5%          Shared libraries
  349903   55.3%          Unaccounted
```